tests/form_params_test.go

```diff
@@ -57,99 +63,153 @@ func TestFormParametersMissingContentLength(t *testing.T) {
 	}
-	if !isOkReq(resp) {
-		t.Errorf("response: want %s, got %s", statusOK, string(resp))
+	var tests = []struct {
+		name string
+		req  []byte
+		want want
+	}{
+		{
+			name: "Two Content-Length Headers",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" +
+				"Content-Length: 13\r\n" +
+				"Content-Length: 13\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+			want: want{
+				contentLen:    "13",
+				queryParamVal: "potato",
+			},
+		},
+		{
+			name: "List of same value in Content-Length Header",
+			req: []byte("POST / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" +
+				"Content-Length: 13, 13\r\n" +
+				"\r\n" +
+				"veggie=potato\r\n" +
+				"\r\n"),
+			want: want{
+				contentLen:    "13",
+				queryParamVal: "potato",
+			},
+		},
 	}
 
-	reqBody = "veggie=potato"
-	postReq = []byte("POST / HTTP/1.1\r\n" + "Host: localhost:8080\r\n" + "Content-Type: application/x-www-form-urlencoded; charset=utf-8\r\n" + "Content-Length: 13, 13\r\n" + "\r\n" + reqBody + "\r\n" + "\r\n")
-	resp, err = requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
-		if contentLen := req.Header["Content-Length"][0]; contentLen != "13" {
-			t.Errorf("Expected Content-Length to be 13 but got %s", contentLen)
-			if val := req.FormValue("veggie"); val != "potato" {
-				t.Errorf("req.FormValue: want potato, got %s", val)
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), test.req, func(req *http.Request) {
+				if contentLen := req.Header["Content-Length"][0]; contentLen != test.want.contentLen {
+					t.Errorf("Expected Content-Length to be 13 but got %s", contentLen)
+				}
+				if val := req.FormValue("veggie"); val != test.want.queryParamVal {
+					t.Errorf("req.FormValue: want potato, got %s", val)
+				}
+			})
+			if err != nil {
+				t.Errorf("MakeRequest(): got err %v, want nil", err)
 			}
-		}
-	})
-	if err != nil {
-		t.Errorf("MakeRequest(): got err %v, want nil", err)
-	}
-	if !isOkReq(resp) {
-		t.Errorf("response: want %s, got %s", statusOK, string(resp))
+			if !isOKReq(resp) {
+				t.Errorf("response: want %s, got %s", statusOKReq, resp)
+			}
+		})
 	}
 
-	postReq = []byte("POST / HTTP/1.1\r\n" + "Host: localhost:8080\r\n" + "Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" + "Content-Length: 13\r\n" + "Content-Length: 12\r\n" + "\r\n" + reqBody + "\r\n" + "\r\n")
-	resp, err = requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
+	postReq := []byte("POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: application/x-www-form-urlencoded; charset=ASCII\r\n" + "Content-Length: 13\r\n" +
+		"Content-Length: 12\r\n" +
+		"\r\n" +
+		"veggie=potato\r\n" +
+		"\r\n")
```

Comment:
Maybe put this in a sub-test (`t.Run`) as well since the other tests are in one.